src/OpenFeature.Contrib.Providers.Flagd/Resolver/InProcess/JsonEvaluator.cs

```diff
@@ -88,7 +83,59 @@ internal FlagSyncData Parse(string flagConfigurations)
                 });
             }
 
-            return JsonConvert.DeserializeObject<FlagSyncData>(transformed);
+
+            var data = JsonConvert.DeserializeObject<FlagSyncData>(transformed);
+            if (data.Metadata == null)
+            {
+                data.Metadata = new Dictionary<string, object>();
+            }
+            else
+            {
+                foreach (var key in new List<string>(data.Metadata.Keys))
+                {
+                    var value = data.Metadata[key];
+                    if (value is long longValue)
+                    {
+                        data.Metadata[key] = (int)longValue;
+                        continue;
+                    }
+
+                    VerifyMetadataValue(key, value);
+                }
+            }
+
+            foreach (var flagConfig in data.Flags)
+            {
+                if (flagConfig.Value.Metadata == null)
+                {
+                    continue;
+                }
+
+                foreach (var key in new List<string>(flagConfig.Value.Metadata.Keys))
+                {
+                    var value = flagConfig.Value.Metadata[key];
+                    if (value is long longValue)
+                    {
+                        flagConfig.Value.Metadata[key] = (int)longValue;
+                        continue;
```

Comment:
[suggestion] Although we know now that based on the current implementation of `VerifyMetaDataValue,` we do have a correct value, what if we extend the method with more checks? we might not check for int, hence i would drop this `continue` and verify all the time
```suggestion
```